dt-workflows/workflows-execution-handler.php

```diff
@@ -583,14 +584,68 @@ private static function action_append( $field_type, $field_id, $value ): array {
                 }
                 break;
             case 'comments':
+
+                $value = self::get_references( $value, $post );
+
+                // create the comment
                 $updated['notes'] = [
                     $value,
                 ];
+
         }
 
         return $updated;
     }
 
+    public static function get_references( $value, $post ) {
+        // get post settings of the current post
+        $settings = DT_Posts::get_post_settings( $post['post_type'] );
+        // filter down to only reference fields - or can you just get connection_types?
+        $fields = array_filter( $settings['fields'], function( $obj ) {
+            if ( $obj['type'] == 'connection' || $obj['type'] == 'user_select' ) {
+                return true;
+            }
+            return false;
+        } );
+
+        $re = '/\{(\S+?)\}/m';
+        $replacement = '';
+        //$str = 'Notify {subassigned} and {assigned_to}';
+
+        preg_match_all($re, $value, $matches, PREG_SET_ORDER, 0);
+
+        // Print the entire match result
+        //var_dump($matches);
+
+        foreach ( $matches as $match ) {
+            $references = $post[$match[1]];
+
+            if ( $fields[$match[1]]['type'] == 'user_select' ) {
+                $replacement = $replacement . '@[' . $references['display'] . '](' . $references['id'] . ')';
+            } else {
+                // otherwise loop through references, building replacement string
+                foreach ( $references as $current => $reference ) {
+                    $id = '';
+                    $reference_post = DT_Posts::get_post( $fields[$match[1]]['post_type'], $reference['ID'], false, false );
+                    // if the post corresponds to a user, @mention that user, otherwise just the post name
+                    if ( !empty( $reference_post['corresponds_to_user'] ) ) {
+                        $id = $reference_post['corresponds_to_user'];
+                        $replacement = $replacement . '@[' . $reference['post_title'] . '](' . $id . ')';
+                    } else {
+                        $replacement = $replacement . $reference['post_title'];
+                    }
+                    // if not the last reference, add a comma & space
+                    if ( $current != sizeof( $references ) ) {
+                        $replacement = $replacement . ', ';
+                    }
+                }
+            }
+
+            // replace the {field} in the comment w/ the references
+            return str_replace( $match[0], $replacement, $value );
```

Comment:
If outer foreach loop is to remain, consider moving return statement outside of loop; to be the last statement within `get_references()` function.